src/SprykerEco/Yves/Payone/Form/DataProvider/CashOnDeliveryDataProvider.php

```diff
@@ -22,7 +22,7 @@ class CashOnDeliveryDataProvider implements StepEngineFormDataProviderInterface
      */
     public function getData(AbstractTransfer $quoteTransfer): QuoteTransfer
     {
-        if ($quoteTransfer->getPayment() === null) {
+        if ($quoteTransfer instanceof QuoteTransfer && $quoteTransfer->getPayment() === null) {
```

Comment:
```suggestion
        if ($quoteTransfer->getPayment() === null) {
```

Reply:
It is a condition for passing the validation of PHPStan and code-style sniffer.